traceevents/src/main/java/com/tomtom/kotlin/traceevents/Tracer.kt

```diff
@@ -294,6 +297,25 @@ class Tracer private constructor(
             return null
         }
 
+        /**
+         * Get the parameter names for the method.
+         * For performance reasons, Make sure this code is executed only once per method call.
+         */
+        val parameterNames: Array<String>
+        if (parameterNamesCache.containsKey(method)) {
+            parameterNames = parameterNamesCache.get(method)!!
+        } else {
+            val kotlinFunction = method.kotlinFunction
+            if (kotlinFunction == null) {
+                parameterNames = arrayOf()
+            } else {
+                parameterNames = kotlinFunction.parameters.filter { it.name != null }.map { it.name!! }.toTypedArray()
```

Comment:
This line worries me a little bit. Is there really no way that `parameterNames` ends up a different length than args? The documentation about when name is null doesn't really sound like it is guaranteed to be the complete list of when it happens. It is a pre-condition for `getParametersMap`. Maybe good to assert that here as well? Or maybe it is a place to be a little defensive and ignore it when it is shorter than args?

Reply:
Good point. I've changed the interface to make sure `getParametersMap` either returns a vallid map, or `null` if the parameter names and args don't match up.